File: GeekBand/python/reverseWords.py
"""
 * @file     reverseWords.py
 * @brief    https://leetcode-cn.com/problems/reverse-words-in-a-string-iii/
 * @author   YongDu
 * @date     2021-08-18
"""


class Solution:
    def reverseWords(self, s: str) -> str:
        return ' '.join(word[::-1] for word in s.split(' '))

File: GeekBand/javascript/reverseWords.js
/**
 * 利用API
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    return s.split(' ').map(w => Array.from(w).reverse().join('')).join(' ');
};

/**
 * 朴素的遍历
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
    let sentence = '';
    let word = '';
    for (const c of s) {
        if (c == ' ') {
            sentence += word + ' ';
            word = '';
        } else {
            word = c + word;
        }
    }
    sentence += word;

    return sentence;
};

/**
 * 一个奇怪的思路，看上去很酷
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
    return s
        .split('')
        .reverse()
        .join('')
        .split(' ')
        .reverse()
        .join(' ');
};